packages/model-viewer/src/features/scene-graph/model.ts

```diff
@@ -23,14 +23,16 @@ import {Material} from './material.js';
 
 
 const $materials = Symbol('materials');
+const $variants = Symbol('variants');
 
 /**
  * A Model facades the top-level GLTF object returned by Three.js' GLTFLoader.
  * Currently, the model only bothers itself with the materials in the Three.js
  * scene graph.
  */
 export class Model implements ModelInterface {
-  private[$materials]: Array<Material> = [];
+  private[$materials]: Material[] = [];
+  private[$variants]: Map<number, Material> = new Map<number, Material>();
```

Comment:
A map of variants materials is created instead of a list as variants are added out of order and the built-in gltf indexing would not work.

Reply:
No need to type the Map; TS can infer from the constructor. So we're mapping a variant number to a material? Don't variants have multiple materials? Or is this some kind of per-mesh variant number?

Reply:
Oh wait, I think I see now; this maps a glTFMaterial index to our Material wrapper object. I think the name  is confusing to me here; I understand this is helpful for variants, but I don't think anything about this data structure is variants-specific, correct? This seems pretty duplicate to our `$materials` array. Would it be possible to stick with the original array but populate it with `null` for any index that  hasn't been  created yet? Or else just remove $materials entirely (though that would be breaking).

Reply:
I was able to collapse into one list of materials by carving out a space for un-instantiated materials. It does require changing the type from `material: Array<Material>` to `material: Array<Material|null>` which created a compile in typescript where the list is accessed and assumed to be not-null, obviously the compiled javascript is the same.